### PR TITLE
Test Example Images Exist - Release Automation

### PR DESCRIPTION
**Pull request by Kalaiselvi84**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

/kind cleanup

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

#2460 

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by markmandel** [comment]
Looks like a flake. Will run again when there is less other stuff in flight.

File: build/release/cloudbuild.yaml [hunk 1]
```diff
@@ -13,12 +13,39 @@
...
+  - name: "make-docker"
+    id: pull-build-image
+    dir: "build"
+    env:
+      - "REGISTRY=us-docker.pkg.dev/${PROJECT_ID}/ci"
+    args: ["pull-build-image"]
+    waitFor:
+      - build-make-docker
+  
+  - name: make-docker
```
Comment (markmandel) [inline build/release/cloudbuild.yaml:34]: This all looks really good!

File: build/includes/examples.mk [hunk 2]
```diff
@@ -42,8 +42,9 @@ push-example-golang-images: example-image-push.supertuxkart
 push-example-golang-images: example-image-push.simple-game-server
 
 # Test to ensure the example image found in the % folder is on GAR. Fails if it is not.
-example-image-test.%:
-	$(DOCKER_RUN) bash -c "cd examples/$* && make gar-check"
+example-image-test.%: 
+	$(DOCKER_RUN) bash -c "cd examples/$* && make gar-check" $(DOCKER_RUN_ARGS)
```
Comment (markmandel) [inline build/includes/examples.mk:46]:
```suggestion
	$(DOCKER_RUN) bash -c "cd examples/$* && make gar-check"
```

🤔  do you need this change? I would have thought the $(DOCKER_RUN_ARGS) would be included in the `$(DOCKER_RUN)` command?

File: build/includes/examples.mk [hunk 3]
```diff
@@ -42,8 +42,9 @@ push-example-golang-images: example-image-push.supertuxkart
 push-example-golang-images: example-image-push.simple-game-server
 
 # Test to ensure the example image found in the % folder is on GAR. Fails if it is not.
-example-image-test.%:
-	$(DOCKER_RUN) bash -c "cd examples/$* && make gar-check"
+example-image-test.%:
```
Comment (aimuz) [inline build/includes/examples.mk:45]:
```suggestion
example-image-test.%:
```
Is there an extra space here?

File: build/release/cloudbuild.yaml [hunk 4]
```diff
@@ -27,19 +54,21 @@ steps:
...
   dynamic_substitutions: true
 substitutions:
   _STORAGE: gs://${PROJECT_ID}-agones-releases
+
 availableSecrets:
   secretManager:
     - versionName: projects/${PROJECT_ID}/secrets/release-github-token/versions/latest
       env: GH_TOKEN
 timeout: 1800s
-
```
Comment (aimuz) [inline build/release/cloudbuild.yaml:45]: Do we have a formatting tool for yaml? @markmandel

File: build/includes/examples.mk (hunk 2 above)
Comment (Kalaiselvi84) [reply build/includes/examples.mk]: Thanks for the review. I have updated this line.

File: build/includes/examples.mk (hunk 3 above)
Comment (Kalaiselvi84) [reply build/includes/examples.mk]: Thanks for the review. I have deleted the extra space.

File: build/includes/examples.mk (hunk 2 above)
Comment (markmandel) [reply build/includes/examples.mk]: Just confirming, you tested it, and it works without the updated line?

**Comment by markmandel** [comment]
Looks good, we should also update our release checklist, since we have a better process now:

This line can be removed, since it's being covered by this update to `build-release`
https://github.com/googleforgames/agones/blob/08c1eaac4552433c9c58bcc9e80c5e16a59788a9/docs/governance/templates/release_issue.md?plain=1#L45

And we should update 
https://github.com/googleforgames/agones/blob/08c1eaac4552433c9c58bcc9e80c5e16a59788a9/docs/governance/templates/release_issue.md?plain=1#L35

To include a note that it's also checking the example images (similar to the above line we are removing above).

File: build/includes/examples.mk (hunk 2 above)
Comment (Kalaiselvi84) [reply build/includes/examples.mk]: Yes, I have tested with agones-mangalpalli. Please find the build log - https://paste.googleplex.com/5228291551133696

Comment (markmandel) [reply build/includes/examples.mk]: Awesome. Thanks for confirming!

File: build/release/cloudbuild.yaml [hunk 5]
```diff
@@ -13,12 +13,39 @@
...
+      - "REGISTRY=us-docker.pkg.dev/${PROJECT_ID}/ci"
+    args: ["pull-build-image"]
+    waitFor:
+      - build-make-docker
+  
+  - name: make-docker
+    id: test-examples-on-gar
+    dir: "build"
+    env:
+    - "REGISTRY= us-docker.pkg.dev/${PROJECT_ID}"
```
Comment (roberthbailey) [inline build/release/cloudbuild.yaml:38]: nit: Is the extra space after the = here necessary?

**Review by markmandel — APPROVED** [review]

**Pull request merged**